src/components/messages/conversation-preview.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { FC } from "react";
+import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import { Conversation, Profile } from "../../models";
+
+const styles = StyleSheet.create({
+  container: {
+    height: 65,
+    justifyContent: "center",
+    marginVertical: 2,
+  },
+  infoContainer: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+    paddingHorizontal: 5,
+  },
+  picture: {
+    height: 60,
+    width: 60,
+    borderRadius: 60 / 2,
+  },
+  titleAndMessageContainer: {
+    flexDirection: "column",
+    alignItems: "flex-start",
+    justifyContent: "center",
+    width: "100%",
+    paddingLeft: 10,
+  },
+  title: {
+    fontWeight: "bold",
+    fontSize: 17,
+    marginBottom: 4,
+  },
+  messageAndTime: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+  },
+  message: {
+    flex: 0.7,
+    fontSize: 14,
+    color: "#9E9E9E",
+  },
+  date: {
+    flex: 0.3,
+    fontSize: 14,
+    color: "#9E9E9E",
+    paddingLeft: 10,
+  },
+});
+
+type Props = {
+  conversation: Conversation;
+  profile: Profile;
+};
+
+export const ConversationPreview: FC<Props> = (props: Props) => {
+  const getTitle: () => string = () => {
+    if (props.conversation.title) return props.conversation.title;
+    const profiles = props.conversation.users.filter(
+      (user) => user.id !== props.profile.id
+    );
+    const fullnames = profiles.map((profile) => {
+      return [profile.firstName, profile.lastName].join(" ");
+    });
+    return fullnames.join(", ");
+  };
+
+  const getPicture: () => string = () => {
+    const profiles = props.conversation.users.filter(
+      (user) => user.id !== props.profile.id
+    );
+    return profiles[0].profilePicture;
+  };
+
+  const getLastMessage: () => string = () => {
+    let lastMessage =
+      props.profile.id ===
+      props.conversation.messages[props.conversation.messages.length - 1].sender
+        .id
+        ? "Vous: "
+        : props.conversation.users.length > 2
+        ? props.conversation.messages[props.conversation.messages.length - 1]
+            .sender.firstName + ": "
+        : "";
+    lastMessage +=
+      props.conversation.messages[props.conversation.messages.length - 1]
+        .content;
+    return lastMessage;
+  };
+
+  const getTime: () => string = () => {
+    const timestamp =
+      props.conversation.messages[props.conversation.messages.length - 1]
+        .timestamp;
+    const date = new Date(+timestamp).toLocaleString();
+    const now = new Date().toLocaleString().slice(0, 10);
+    return date.slice(0, 10) !== now ? date.slice(0, 5) : date.slice(12, 17);
+  };
```

Comment:
I did not understand the purpose of this function (in particular the ternary)

Here is the result when the timestamp is the same and when the timestamp is not the same: https://repl.it/talk/share/ThoseEmbellishedParallelcomputing/114822

Reply:
This function aims to get the date of the last message. If the message was sent today we get the hour like HH/MM and if the last message was sent another day we display the date like DD/MM. .toLocaleString() get a readable format of the timestamp and we extract one part or another to display the date. So the bug with the date may come from the result of this function that may defer on Android and IOS devices. But I can't launch an Android device on my computer so I can't check it :(